dbt/adapters/base/relation.py

```diff
@@ -531,3 +531,10 @@ def flatten(self, allow_multiple_databases: bool = False) -> "SchemaSearchMap":
             )
 
         return new
+
+
+@dataclass(frozen=True, eq=False, repr=False)
+class AdapterTrackingRelationInfo(FakeAPIObject, Hashable):
```

Comment:
This is an interesting case to me: should this actually be considered a "contract" like `AdapterResponse` or is it tied to the concept of a relation (i.e. the thing it's tracking)? If the former we should probably put it in it's own module in `dbt/adapters/contracts`. I think there's an argument for both so curious what others think

Reply:
So just putting this forward -- I originally had it as a contract. Then it ran into "can't instantiate protocol" errors. So I'm not sure it should be a contract on that alone :p